src/AvPortalClient.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\media_avportal;
+
+use Drupal\Component\Serialization\Json;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use GuzzleHttp\ClientInterface;
+
+/**
+ * Client that interacts with the AV Portal.
+ */
+class AvPortalClient {
+
+  /**
+   * The HTTP client.
+   *
+   * @var \GuzzleHttp\ClientInterface|\GuzzleHttp\Client
+   */
+  protected $httpClient;
+
+  /**
+   * The module configuration.
+   *
+   * @var \Drupal\Core\Config\ImmutableConfig
+   */
+  protected $config;
+
+  /**
+   * Constructs an AvPortalClient object.
+   *
+   * @param \GuzzleHttp\ClientInterface $http_client
+   *   The HTTP client.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   The config factory.
+   */
+  public function __construct(ClientInterface $http_client, ConfigFactoryInterface $configFactory) {
+    $this->httpClient = $http_client;
+    $this->config = $configFactory->get('media_avportal.settings');
+  }
+
+  /**
+   * Executes the query call for resources.
+   *
+   * @param array $options
+   *   The options.
+   *
+   * @return array|null
+   *   The array if the query succeeded, NULL otherwise.
+   *
+   * @throws \Exception
+   */
+  public function query(array $options): ?array {
+    $options += [
+      'fl' => 'type,ref,doc_ref,titles_json,duration,shootstartdate,media_json,mediaorder_json,summary_json',
+      'hasmedia' => 1,
+      'wt' => 'json',
+      'index' => 1,
+      'pagesize' => 15,
+      'type' => 'VIDEO',
+    ];
+
+    $response = $this->httpClient->get($this->config->get('client_api_uri'), ['query' => $options]);
+    $response = Json::decode((string) $response->getBody())['response'];
+
+    return $this->resourcesFromResponse($response);
+  }
+
+  /**
+   * Returns a single resource.
+   *
+   * @param string $ref
+   *   The reference.
+   *
+   * @return \Drupal\media_avportal\AvPortalResource|null
+   *   The resource.
+   *
+   * @throws \Exception
+   */
+  public function getResource(string $ref):? AvPortalResource {
+    $result = $this->query(['ref' => $ref]);
+
+    $result['resources'] += [$ref => NULL];
```

Comment:
Instead of using this, we should go with a ternary operator: `return $result['resources'][$ref] ?? NULL;`.